### DISCUSSION TITLE
Language detected is spanish instead of english -> the transcription is translation of english to spanish

### DISCUSSION THREAD

**Discussion by dan19** [deepgram/community#236] [General help]
### Which Deepgram product are you using?

Deepgram API

### Details

Hello,

I've been seeing cases over the past week where the automatic language detection detects "es" instead of "en" and then generate a transcription that seems to be the translation of the english content in Spanish.
I'm using `whisper-large` and this is the params I'm sending:
```
{
    model: 'whisper-large',
    detect_language: 'true',
    punctuate: 'true',
    channels: '1',
    callback: <callbackUrl>
}
```

### If you are making a request to the Deepgram API, what is the full Deepgram URL you are making a request to?

https://api.deepgram.com/v1/listen?smart_format=true&detect_language=true&model=whisper-large&punctuate=true&channels=1

### If you are making a request to the Deepgram API and have a request ID, please paste it below:

3f1488ab-6b75-4809-b372-d1c5fbdbc2ac

**Comment by jjmaldonis** [comment]
Hey @dan19 that's very surprising! Do you have an example audio file? We'd like to reproduce the result.

**Reply by dan19** [reply]
@jjmaldonis can I share an audio file with you privately?

**Reply by jjmaldonis** [reply]
Yes absolutely. Please send me an email at

**Reply by jjmaldonis** [reply]
Thanks for the audio example! I was able to reproduce the issue. If I specify the language manually, then the transcription works as intended without translating. This means the detect-language feature is not correctly identifying the language for this audio. I'll need to follow up with our team to ask if there is any way to improve the detect-language results.

Below is the code I'm using to test this out:

```
from deepgram import Deepgram
import json
import os


DEEPGRAM_API_KEY = os.environ["DEEPGRAM_API_KEY"]  # Your Deepgram API Key
deepgram = Deepgram(DEEPGRAM_API_KEY)


def transcribe_url(url: str):
    options = {
        "model": "whisper-large",
        # "detect_language": True,
        "language": "en-US",  # `en-US` translates the audio, and `fr` does not
        "punctuate": True,
        "channels": 1,
    }
    source = {"url": url}
    result = deepgram.transcription.sync_prerecorded(source, options)
    print(json.dumps(result, indent=2))


if __name__ == "__main__":
    url = "..."  # French audio
    transcribe_url(url)
```
